### DISCUSSION TITLE
How do you structure your routes folder with new RR7 `routes` file

### DISCUSSION THREAD

**Discussion by MadadiSaeed** [remix-run/react-router#12632] [Q&A] [answered]
Hi since recommended way of defining routes in RR7 is using `routes` file I am curious about your suggestion about  the structure of the routes folder. What do you recommend? Do you prefer flat folder structure or nested folder structure or any other suggestions?

**Comment by sergiodxa** [comment, marked as answer]
I either keep using the [file route convention](https://reactrouter.com/how-to/file-route-conventions) from Remix v2 or I like to organize route files like this:

First `app/resources` for all resource routes (routes without UI) like API routes, webhooks, or other things. that are inside `app/resources` directly or inside other folders as needed.

```
app/resources          # Resource Routes
app/resources/api/     # API Routes
app/resources/webhooks # Webhook Routes
```

Then `app/views` for all routes with a UI where I organize it like

```
app/views                      # UI Routes
app/views/layouts              # Layout Routes
app/views/[resource]/index.tsx # The list routes of a resource
app/views/[resource]/show.tsx  # The detail route
app/views/[resource]/new.tsx   # The route to create a new resource
app/views/[resource]/edit.tsx  # The route to edit a resource
```

The layouts will only have files app auth.tsx, app.tsx, landing.tsx and other layouts that will be shared.

The [resource] is replaced with something like users, posts, comments, basically like a REST API, so inside you have an index to list them, a show for a detail, and new and edit to create/update.

Then in `app/routes.ts` I can use that to create any URL I want.

In this convention the file doesn't relate to the URL, URLs are defined in `app/routes.ts` not in the file system.

This convention is similar to what frameworks like Ruby on Rails does.

**Comment by MadadiSaeed** [comment]
Thank you so much for the response. Very interesting view. I like this approach.
